torch/_dynamo/skipfiles.py

```diff
@@ -62,58 +137,39 @@ def _module_dir(m: types.ModuleType):
     return _strip_init_py(m.__file__)
 
 
-SKIP_DIRS = [
-    # torch.*
-    _module_dir(torch),
-    # torchdynamo.*
-    os.path.dirname(__file__) + "/",
-    "<frozen importlib",
-    "<__array_function__ internals>",
-] + [
-    # skip some standard libs
-    _module_dir(m)
-    for m in (
-        abc,
-        collections,
-        contextlib,
-        copy,
-        copyreg,
-        dataclasses,
-        enum,
-        functools,
-        importlib,
-        inspect,
-        linecache,
-        logging,
-        multiprocessing,
-        operator,
-        os,
-        posixpath,
-        random,
-        re,
-        selectors,
-        signal,
-        tempfile,
-        threading,
-        tokenize,
-        traceback,
-        types,
-        typing,
-        unittest,
-        weakref,
-        _collections_abc,
-        _weakrefset,
-    )
-]
-
+# Force inline functions in these files, even the files is in *_SKIPLIST.
 FILENAME_ALLOWLIST = {
     torch.nn.Sequential.__init__.__code__.co_filename,
     torch.set_rng_state.__code__.co_filename,
     torch._inductor.test_operators.__file__,
     torch.utils._content_store.__file__,
-    # These are dynamo files!
     external_utils.__file__,
-    comptime.__file__,  # Want to inline these helpers
+    comptime.__file__,
+    torch.optim._functional.__file__,
+    torch.utils._foreach_utils.__file__,
```

Comment:
Why not do these consistently with below?

Reply:
Legacy code, we should make them consistent. But I hit some weird error now, so keep as is. Will address later or in a separate PR.